src/pages/storage/StorageVolumeNameLink.tsx

```diff
@@ -23,21 +25,44 @@ const StorageVolumeNameLink: FC<Props> = ({
 }) => {
   const { project } = useCurrentProject();
   const isExternalLink = !hasVolumeDetailPage(volume);
-  const caption = overrideName ? overrideName : volume.name;
+
+  const isInstance =
+    volume.type === "container" || volume.type === "virtual-machine";
+  const { data: instances = [] } = useInstances(volume.project);
+  const instance =
+    isInstance && instances.find((instance) => instance.name === volume.name);
+
+  const isImage = volume.type === "image";
+  const { data: images = [] } = useImagesInProject(volume.project);
+  const image =
+    isImage && images.find((image) => image.fingerprint === volume.name);
+
+  const isCustom = volume.type === "custom";
+  const displayLink = instance || image || isCustom;
+
+  const caption = overrideName
+    ? displayLink
+      ? overrideName
+      : ""
+    : volume.name;
```

Comment:
If I understand correctly, this is to not render anything in the right-hand column on the volume list. Currently, we still render some markup with titles and empty string as content in the case that link should be hidden. Better simplify and avoid the empty markup, as suggested below?

Might be worth adding a comment or solving this in a more straight forward way to make it easy to understand.

```suggestion
  const caption = overrideName ? overrideName : volume.name;
  
  if (overrideName && !displayLink) {
    return null;
  }
```